src/groth16/prover.rs

```diff
@@ -275,8 +276,28 @@ where
 
     // Preparing things for the proofs is done a lot in parallel with the help of Rayon. Make
     // sure that those things run on the correct thread pool.
-    let (start, mut provers, input_assignments, aux_assignments) =
-        THREAD_POOL.install(|| create_proof_batch_priority_inner(circuits))?;
+    let mut provers = THREAD_POOL.install(|| {
+        circuits
+            .into_par_iter()
+            .map(|circuit| -> Result<_, SynthesisError> {
+                let mut prover = ProvingAssignment::new();
+
+                prover.alloc_input(|| "", || Ok(E::Fr::one()))?;
+
+                circuit.synthesize(&mut prover)?;
+
+                for i in 0..prover.input_assignment.len() {
+                    prover.enforce(|| "", |lc| lc + Variable(Index::Input(i)), |lc| lc, |lc| lc);
+                }
+
+                Ok(prover)
+            })
+            .collect::<Result<Vec<_>, _>>()
+    })?;
```

Comment:
Closing the `THREAD_POOL.install()` should happen at [line 610](https://github.com/filecoin-project/bellperson/pull/141/files?diff=split&w=1#diff-45d555fd2845f5f7080c2fb2b021dcdcd3110df42fd6e378c3780275dec69344R610). Else the `par!{}` call, else things would run on some global Rayon thread pool and not on the `THREAD_POOL`.

Reply:
fixed